src/Dhall.hs

```diff
@@ -103,6 +105,12 @@ instance Show InvalidType where
 
 instance Exception InvalidType
 
+-- | Dhall uses this type to represent (a, b) tuple.
+data R2 a b = R2 { _1 :: a, _2 :: b } deriving (Show, Generic)
+
+instance (Inject a, Inject b) => Inject (R2 a b)
```

Comment:
If you add the `DeriveAnyClass` extension, then you can simplify this further as just:

```haskell
data R2 a b = R2 { _1 :: a, _2 :: b }
    deriving (Generic, Inject, Interpret, Show)
```